src/A11yInterface.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+A11yInterface.propTypes = {
+  /**
+   * A function that takes the index of the frame,
+   * and returns a string to render as an aria label
+   * for the specific frame in the Interface.
+   *
+   * (frameIndex) => string
+   */
+  ariaLabelGenerator: PropTypes.func.isRequired,
+  /**
+   * the number of frames to render across the visualization
+   * (e.g. the number of data points to message via screen reader)
+   */
+  numFrames: PropTypes.number,
+  /**
+   * an optional `onKeyDown` event handler to provide for each frame
+   * when selected.
+   * Best practices are to describe how the user will interact with the chart in
+   * the ariaLabelGenerator
+   *
+   * (event, frameIndex) => void
+   */
+  onKeyDown: PropTypes.func,
+  /**
+   * height of the chart - provided by `XYPlot`
+   */
+  height: PropTypes.number,
+  /**
+   * width of the chart - provided by `XYPlot`
+   */
+  width: PropTypes.number,
+};
+
+/**
+ * `A11yInterface` provides a mechanism for keyboard navigation of
+ * `LineChart` and `AreaChart` graphs within `XYPlot`. This component renders
+ * a rectangle for a given `numFrames` across a Line/Area chart. `numFrames` should
+ * be equal to the number of data points in your dataset. Users can navigate to a frame
+ * either by pressing `Tab`, or through their assistive technology. `A11yInterface`
+ * takes an `ariaLabelGenerator`, which generates an aria-label for each frame of the chart.
+ * [aria-labels](https://www.w3.org/TR/WCAG20-TECHS/ARIA6.html#ARIA6-description) are critical for users accessing
+ * the web with screenreaders or other assistive technologies.
+ */
+
```

Comment:
I really appreciate the detailed comment here.

I'm thinking of a few other ways to tweak the API here to make it more like other Reactochart functions. Why not just pass `data` as a prop to `A11yInterface` and make `ariaLabel` a function that takes a datum and an optional index, similar to most D3 functions? Then `numFrames` (which I would rename `numLabels`) could default to `data.length` and probably doesn't even have to be passed in. It seems like most of the implementations of ariaLabelGenerator need the data anyway.

Reply:
So I thought a lot about using data. It becomes a little more complex when dealing with multiple data sets. For example, if you have two line charts in an `XYPlot`, you would only want one `A11yInterface`, and that `A11yInterface` would need to account for both data sets. This makes calculating the number of frames a little complicated since `data1` and `data2` can be different lengths, and can contain overlapping values.

I wanted to avoid looping over the data to find all unique data points, though since `lodash` is a dependency here, I suppose we could find the union of `x` values more easily. `unionBy` seems promising and using Reactochart's passed in `dataset` prop, this could work. However, it would make generating the `aria-labels` a little different, since then we'd be working with the `x` value, as opposed to the `frameIndex`. This would also make the Interface more robust with data sets of unequal lengths and unequal `x` values.